test/integration/test_user_defined_output_plugins.py

```diff
@@ -0,0 +1,185 @@
+"""Tests Nighthawk's user defined output framework."""
+
+import json
+import logging
+import math
+import os
+import pytest
+import subprocess
+import sys
+import time
+from threading import Thread
+
+from test.integration.common import IpVersion
+from test.integration.integration_test_fixtures import (http_test_server_fixture, server_config)
+from test.integration import asserts
+from test.integration import utility
+
+
+def getUserDefinedOutputsFromJson(parsed_json):
+  """Get the user defined outputs indexed by the result they came from."""
+  return {result["name"]: result["user_defined_outputs"] for result in parsed_json["results"]}
+
+
+@pytest.mark.parametrize('server_config',
+                         ["nighthawk/test/integration/configurations/nighthawk_http_origin.yaml"])
+def test_all_plugin_apis_called(http_test_server_fixture):
+  """Checks that a User Defined Output Plugin produces correct output."""
+  fake_plugin_config = (
+      "{name:\"nighthawk.fake_user_defined_output\","
+      "typed_config:{\"@type\":\"type.googleapis.com/nighthawk.FakeUserDefinedOutputConfig\"}}")
+
+  parsed_json, _ = http_test_server_fixture.runNighthawkClient([
+      http_test_server_fixture.getTestServerRootUri(), "--duration", "100", "--concurrency", "2",
+      "--termination-predicate", "benchmark.http_2xx:24", "--rps", "100",
+      "--user-defined-plugin-config", fake_plugin_config
+  ])
+  counters = http_test_server_fixture.getNighthawkCounterMapFromJson(parsed_json)
+  asserts.assertCounterGreaterEqual(counters, "benchmark.http_2xx", 50)
+
+  user_defined_outputs_by_result = getUserDefinedOutputsFromJson(parsed_json)
+  asserts.assertEqual(len(user_defined_outputs_by_result), 3)
+
+  for result_name in user_defined_outputs_by_result:
```

Comment:
Optional: 
The key and value can be retrieved at the same time from a dictionary if so desired.
  _for k, v in dict.items():
     ..._

Applies below as well.

Reply:
Done. Thanks for the python knowledge!